lib/cn/kblock_builder.c

```diff
@@ -1188,9 +1208,9 @@ kbb_finish(struct kblock_builder *bld, struct blk_list *kblks, u64 seqno_min, u6
      */
     if (!kblock_is_empty(&bld->curr)) {
         struct wbb *pt = 0;
-        u64         kbsize = bld->rp->kblock_size;
-        u64         ptsize = (wbb_page_cnt_get(bld->ptree)) * PAGE_SIZE;
-        u64         kbused =
+        uint32_t    kbsize = bld->max_size;
+        uint64_t    ptsize = (wbb_page_cnt_get(bld->ptree)) * PAGE_SIZE;
+        uint64_t    kbused =
```

Comment:
Reducing the size of variables runs the risk of quietly truncating the size of a calculation that was previously carried out in 64-bit math to 32-bit math.  Tread very very carefully..